### PR TITLE
Reorganise e2e tests

### PR DESCRIPTION
**Pull request by mapno**
**What this PR does**:

Moves some of the heaviests e2e tests to their own folder, so they're more organised and can be called independently—e2e tests have been timing out lately.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -119,6 +119,9 @@ jobs:
       - name: Poller
         run: make test-integration-poller
 
+      - name: Deployment modes
```
Comment (javiermolinar) [inline .github/workflows/ci.yml:122]: I'm wondering if we could take the poller and deployment tests outside as a job to see if they can run at the same time

**Review by javiermolinar — COMMENTED** [review]
Cool!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (javiermolinar) [reply .github/workflows/ci.yml:122]: Trying that here: https://github.com/grafana/tempo/pull/3928

**Comment by zalegrala** [comment]
The change looks reasonable to me, but I see that it still took 29 minutes to run.  Is that an improvement in the duration?

**Comment by mapno** [comment]
> The change looks reasonable to me, but I see that it still took 29 minutes to run. Is that an improvement in the duration?

Not really, the main goal was to circumvent the test timeout.

I've updated the PR to split the different targets into different jobs

**Comment by mapno** [comment]
Required CI checks need to be reviewed after this PR gets merged.

**Review by javiermolinar — APPROVED** [review]
Great!

**Review by electron0zero — APPROVED** [review]

**Pull request merged**